src/main/java/com/mparticle/kits/AppseeKit.java

```diff
@@ -0,0 +1,219 @@
+
+        List<ReportingMessage> messageList = new LinkedList<>();
+        messageList.add(
+                new ReportingMessage(this, ReportingMessage.MessageType.OPT_OUT, System.currentTimeMillis(), null)
+                        .setOptOut(optedOut)
+        );
+        return messageList;
+    }
+
+    @Override
+    public List<ReportingMessage> leaveBreadcrumb(String s) {
+        return null;
+    }
+
+    @Override
+    public List<ReportingMessage> logError(String s, Map<String, String> map) {
+        return null;
+    }
+
+    @Override
+    public List<ReportingMessage> logException(Exception e, Map<String, String> map, String s) {
+        return null;
+    }
+
+    @Override
+    public List<ReportingMessage> logEvent(MPEvent mpEvent) {
+        Map<String, Object> eventProps = new HashMap<>();
+
+        eventProps.put("type", mpEvent.getEventType().name());
+
+        String category = mpEvent.getCategory();
+        if (category != null && !category.isEmpty()) {
+            eventProps.put("category", category);
+        }
+
+        Map<String, String> info = mpEvent.getInfo();
+        if (info != null) {
+
+            // Remove the category (if exists) since we already fetched it from the property of the MPEvent
+            if (info.containsKey("$Category")) {
+                info.remove("$Category");
+            }
+
+            if (info.size() > 0) {
+                eventProps.putAll(info);
+            }
+        }
+
+        Appsee.addEvent(mpEvent.getEventName(), eventProps);
+
+        List<ReportingMessage> messageList = new LinkedList<>();
+        messageList.add(ReportingMessage.fromEvent(this, mpEvent));
+        return messageList;
+    }
+
+    @Override
+    public List<ReportingMessage> logScreen(String s, Map<String, String> map) {
+        Appsee.startScreen(s);
+
+        List<ReportingMessage> messageList = new LinkedList<>();
+        messageList.add(
+                new ReportingMessage(this, ReportingMessage.MessageType.SCREEN_VIEW, System.currentTimeMillis(), null)
+                        .setScreenName(s));
+        return messageList;
+    }
+
+    @Override
+    public void onIdentifyCompleted(MParticleUser mParticleUser, FilteredIdentityApiRequest filteredIdentityApiRequest) {
+
+    }
+
+    @Override
+    public void onLoginCompleted(MParticleUser mParticleUser, FilteredIdentityApiRequest filteredIdentityApiRequest) {
+
+    }
+
+    @Override
+    public void onLogoutCompleted(MParticleUser mParticleUser, FilteredIdentityApiRequest filteredIdentityApiRequest) {
+
+    }
+
+    @Override
+    public void onModifyCompleted(MParticleUser mParticleUser, FilteredIdentityApiRequest filteredIdentityApiRequest) {
+
+    }
+
+    @Override
+    public void onUserIdentified(MParticleUser mParticleUser) {
+        Map<MParticle.IdentityType, String> userIdentities = mParticleUser.getUserIdentities();
+        for (MParticle.IdentityType identityType : userIdentities.keySet()) {
```

Comment:
Rather than iterating can you just grab it right out of the map?

```
Map<MParticle.IdentityType, String> userIdentities = mParticleUser.getUserIdentities();
if (userIdentities.containsKey(MParticle.IdentityType.CustomerId)) {
    Appsee.setUserId(userIdentities.get(MParticle.IdentityType.CustomerId));
}
```